lib/trento_web/controllers/v1/host_controller.ex

```diff
@@ -28,6 +33,31 @@ defmodule TrentoWeb.V1.HostController do
     render(conn, "hosts.json", hosts: hosts)
   end
 
+  operation :delete,
+    summary: "Deregister a host",
+    description: "Deregister a host agent from Trento",
+    parameters: [
+      id: [
+        in: :path,
+        required: true,
+        type: %OpenApiSpex.Schema{type: :string, format: :uuid}
+      ]
+    ],
+    responses: [
+      no_content: "The host has been deregistered",
+      not_found: TrentoWeb.OpenApi.Schema.NotFound.response(),
+      unprocessable_entity: OpenApiSpex.JsonErrorResponse.response()
+    ]
+
+  @spec delete(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def delete(conn, %{id: host_id}) do
+    case Hosts.deregister_host(host_id) do
```

Comment:
I have some doubts if this `case` wouldn't allow to use the fallback controller for the 404 scenario.
Does this send the expected `Not found` schema? https://github.com/trento-project/web/blob/2.0.0/lib/trento_web/views/error_view.ex#L26

If not, we would need to have something like:
```
with :ok Hosts.deregister_host(host_id) do
    send_resp(conn, 204, "")
else
    {:error, :host_alive} -> send_resp(conn, 422, "")
end
```
And the fallback controller would take care of the not found

Reply:
Just ran this manually and get the JSON schema:
```
curl -v -X DELETE http://localhost:4000/api/v1/hosts/240f96b1-8d26-53b7-9e99-ffb0f2e735b6
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000 (#0)
> DELETE /api/v1/hosts/240f96b1-8d26-53b7-9e99-ffb0f2e735b6 HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/7.82.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< cache-control: max-age=0, private, must-revalidate
< content-length: 85
< content-type: application/json; charset=utf-8
< date: Tue, 06 Jun 2023 07:10:29 GMT
< server: Cowboy
< x-request-id: F2XVe8nL9JnA_gsAAMgG
< 
* Connection #0 to host localhost left intact
{"errors":[{"detail":"The requested resource cannot be found.","title":"Not Found"}]}
```

Reply:
It breaks a bit how other controllers functions are done (using the `with`), as you explicitily return those errors, but well, it works hehe